src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -80,20 +80,24 @@ const UsersFragment = ({
     });
   };
 
-  const exhangeFollower = (newUserLists, willFollow, index) => {
-    if (newUserLists[index].user) {
-      newUserLists[index].user.following = !!willFollow;
-      newUserLists[index].user.user_id_follower = myId;
+  const exhangeFollower = (newUserLists, willFollow, userId) => {
+    const indexUser = newUserLists.findIndex((item) =>
+      item.user ? item.user.user_id === userId : item.user_id === userId
+    );
+    if (newUserLists[indexUser].user) {
+      newUserLists[indexUser].user.following = !!willFollow;
+      newUserLists[indexUser].user.user_id_follower = myId;
     } else {
-      newUserLists[index].following = !!willFollow;
-      newUserLists[index].user_id_follower = myId;
+      newUserLists[indexUser].following = !!willFollow;
+      newUserLists[indexUser].user_id_follower = myId;
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `exhangeFollower` function has been updated to find the user by their `userId` instead of their index in the array. This is a good change as it makes the function more robust and less prone to errors due to changes in the order of users. However, there's no error handling for when `findIndex` doesn't find a match and returns `-1`. If `userId` does not exist in `newUserLists`, `indexUser` will be `-1` and accessing `newUserLists[-1]` will throw an error.
```diff
+    if (indexUser === -1) {
+      console.error(`User with id ${userId} not found`);
+      return;
+    }
     if (newUserLists[indexUser].user) {
       newUserLists[indexUser].user.following = !!willFollow;
       newUserLists[indexUser].user.user_id_follower = myId;
     } else {
       newUserLists[indexUser].following = !!willFollow;
       newUserLists[indexUser].user_id_follower = myId;
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->